app/api/dao/user.py

```diff
@@ -188,3 +189,53 @@ def authenticate(username_or_email, password):
             return user
 
         return None
+
+    @staticmethod
+    def get_achievements(user_id):
+        user = UserModel.find_by_id(user_id)
+        all_relations = user.mentor_relations + user.mentee_relations
+        tasks = []
+        for relation in all_relations:
+            tasks += relation.tasks_list.tasks
+        achievements = list(filter(lambda x: x.get("is_done"), tasks))
+        return achievements
+
+    @staticmethod
+    def get_user_statistics(user_id):
+        user = UserModel.find_by_id(user_id)
+        all_relations = user.mentor_relations + user.mentee_relations
+        (pending_requests, accepted_requests, rejected_requests, completed_relations, cancelled_relations) = (0, 0, 0, 0, 0)
+        for relation in all_relations:
+            if relation.state == MentorshipRelationState.PENDING:
+                pending_requests += 1
+            elif relation.state == MentorshipRelationState.ACCEPTED:
+                accepted_requests += 1
+            elif relation.state == MentorshipRelationState.REJECTED:
+                rejected_requests += 1
+            elif relation.state == MentorshipRelationState.COMPLETED:
+                completed_relations += 1
+            elif relation.state == MentorshipRelationState.CANCELLED:
+                cancelled_relations += 1
+
+        achievements = UserDAO.get_achievements(user_id)
+        if achievements:
+            # We only need the first three of these achievements
+            achievements = achievements[0:3]
+            sorted(achievements, key=lambda x: x.get("created_at"))
+
+        response = {
+            'name': user.name,
+            'pending_requests': pending_requests,
+            'accepted_requests': accepted_requests,
+            'rejected_requests': rejected_requests,
+            'completed_relations': completed_relations,
+            'cancelled_relations': cancelled_relations,
+            'achievements': achievements
+        }
+        return response
+
+    @staticmethod
+    def verify_user(user_id):
```

Comment:
why is this method here? Were you testing something? Please remove this, since its being done by another function

Reply:
I'm sorry, this should not have been in there. Yes, I was testing something. I'll remove it.